rerun_py/tests/api_sandbox/rerun_draft/catalog.py

```diff
@@ -185,9 +188,31 @@ def schema(self) -> Any:
     def segment_ids(self) -> list[str]:
         return self._inner.partition_ids()
 
-    def segment_table(self) -> datafusion.DataFrame:
+    def segment_table(
+        self, join_meta: datafusion.DataFrame | None = None, join_key: str = "rerun_segment_id"
+    ) -> datafusion.DataFrame:
```

Comment:
I really like this! (Also goes to show how powerful slapping some python code on top of internal object can be.)

Reply:
worth having a knob to control the case where some segments are missing in the metadata table? I can see both "fill nulls" and "drop rows" being contextually useful. Footgun could be avoided by defaulting to "strict" mode that would fail on a segment missing a matching  metadata row.